revolv/payments/models.py

```diff
@@ -23,63 +36,92 @@ class PaymentInstrumentType(models.Model):
     objects = PaymentInstrumentTypeManager()
 
 
-class PaymentTransactionManager(models.Manager):
+class PaymentManager(models.Manager):
+    """
+        Simple manager for the Payment model.
+    """
+
+    def payments(self, user=None, project=None, queryset=None):
+        """
+        :return: Payments associated with this user
+        """
+        if queryset is None:
+            queryset = super(PaymentManager, self).get_queryset()
+        payments = queryset.filter(user=user).order_by('created_at')
+        return payments
+
+    def donations(self, user=None, project=None, queryset=None):
+        """
+        :return: Returns all payments that are not repayments or reinvestments
+                 associated with this user.
+        """
+        if queryset is None:
+            queryset = super(PaymentManager, self).get_queryset()
```

Comment:
Yeah this is contestable - the or equals pattern is very common in Ruby so
that's why I mention it.
ᐧ

On Sat, Feb 21, 2015 at 1:32 PM, Noah Gilmore <notifications@github.com>
wrote:

> In revolv/payments/models.py
> <https://github.com/calblueprint/revolv/pull/144#discussion_r25126036>:
>
> > +    def payments(self, user=None, project=None, queryset=None):
> > +        """
> > +        :return: Payments associated with this user
> > +        """
> > +        if queryset is None:
> > +            queryset = super(PaymentManager, self).get_queryset()
> > +        payments = queryset.filter(user=user).order_by('created_at')
> > +        return payments
> > +
> > +    def donations(self, user=None, project=None, queryset=None):
> > +        """
> > +        :return: Returns all payments that are not repayments or reinvestments
> > +                 associated with this user.
> > +        """
> > +        if queryset is None:
> > +            queryset = super(PaymentManager, self).get_queryset()
>
> I would actually argue that the two line if statement rather than the one
> line or is easier to understand from a code readability perspective
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/calblueprint/revolv/pull/144/files#r25126036>.
>



-- 
Sam Lau
EECS | UC Berkeley | May 2017
(562) 565-4336 | samlau95@berkeley.edu